### DISCUSSION TITLE
Introducing: Automatic translations via doloc.io

### DISCUSSION THREAD

**Discussion by daniel-sc** [daniel-sc/ng-extract-i18n-merge#115] [Announcements]
Hi everybody,

I created a service ([doloc.io](https://doloc.io)) which integrates well with `ng-extract-i18n-merge` - you send your merged translation xliff file to `https://api.doloc.io` and **get back a translated version**. As the translation takes already translated texts as a dictionary, these **new translations will be very good** (let me know if not!).

Please check it out and **give some feedback** how it works for you :)

To get started, you could simply create a new npm script, such as (after registering and creating an api token on https://doloc.io/account):

`package.json`:
```diff
  "scripts": {
    "ng": "ng",
    "start": "ng serve",
    "extract-i18n-merge": "ng extract-i18n",
+   "auto-translate-fr": "curl https://api.doloc.io -H \"Authorization: Bearer $API_TOKEN\" --data-binary @src/i18n/messages.fr.xlf -o src/i18n/messages.fr.xlf",
+   "auto-translate-it": "curl https://api.doloc.io -H \"Authorization: Bearer $API_TOKEN\" --data-binary @src/i18n/messages.it.xlf -o src/i18n/messages.it.xlf",
+   "update-i18n": "ng extract-i18n && npm run auto-translate-fr && npm run auto-translate-it",
  },
```

**Comment by ElementalSystems** [comment]
@daniel-sc I believe this script is out of date and you need -H "Authorization: Bearer $API_TOKEN" to make it work now?

**Reply by daniel-sc** [reply]
@ElementalSystems yes - you are right - I updated it accordingly. Thanks!

**Reply by ElementalSystems** [reply]
Still not right - need to escape the " in the json.
 "auto-translate-fr": "curl https://api.doloc.io -H \\"Authorization: Bearer $API_TOKEN\\" --data-binary @src/i18n/messages.fr.xlf -o src/i18n/messages.fr.xlf",

**Reply by daniel-sc** [reply]
@ElementalSystems Thanks again! :)
